File: src/main/java/com/myudog/myulib/api/shape/IShape.java
package com.myudog.myulib.api.shape;
import net.minecraft.world.phys.Vec3;

import java.util.List;
public interface IShape {
    List<Vec3> getOutlinePoints(Vec3 size, double density);
    List<Vec3> getSolidPoints(Vec3 size, double density);
}

File: src/main/kotlin/com/myudog/myulib/api/shape/IShape.kt
package com.myudog.myulib.api.shape

import net.minecraft.util.math.Vec3d

interface IShape {
    /** 獲取邊框點 */
    fun getOutlinePoints(size: Vec3d, density: Double): List<Vec3d>

    /** 獲取內部填充點 (實心) */
    fun getSolidPoints(size: Vec3d, density: Double): List<Vec3d>
}